tools/_venv_common.py

```diff
@@ -8,11 +8,94 @@
 import time
 import subprocess
 import sys
+import re
+
+VERSION_PATTERN = re.compile(r'^(\d+)\.(\d+).*$')
+
+
+class PythonExecutableNotFoundError(Exception):
+    pass
+
+
+def find_python_executable(python_major):
+    # type: (int) -> str
+    """
+    Find the relevant python executable that is of the given python major version.
+    Will test, in decreasing priority order:
+    * the current Python interpreter
+    * 'pythonX' executable in PATH (with X the given major version) if available
+    * 'python' executable in PATH if available
+    * Windows Python launcher 'py' executable in PATH if available
+    Incompatible python versions for Certbot will be evicted (eg. Python < 3.5 on Windows)
+    :param int python_major: the Python major version to target (2 or 3)
+    :rtype: str
+    :return: the relevant python executable path
+    :raise RuntimeError: if no relevant python executable path could be found
+    """
+    python_executable_path = None
+
+    # First try, current python executable
+    if _check_version('{0}.{1}.{2}'.format(
+            sys.version_info[0], sys.version_info[1], sys.version_info[2]), python_major):
+        return sys.executable
+
+    # Second try, with python executables in path
+    versions_to_test = ['2.7', '2', ''] if python_major == 2 else ['3', '']
+    for one_version in versions_to_test:
+        try:
+            one_python = 'python{0}'.format(one_version)
+            output = subprocess.check_output([one_python, '--version'],
+                                             universal_newlines=True, stderr=subprocess.STDOUT)
+            if _check_version(output.strip().split()[1], python_major):
+                return subprocess.check_output([one_python, '-c',
+                                                'import sys; sys.stdout.write(sys.executable);'],
+                                               universal_newlines=True)
+        except (subprocess.CalledProcessError, OSError):
+            pass
+
+    # Last try, with Windows Python launcher
+    try:
+        env_arg = '-{0}'.format(python_major)
+        output_version = subprocess.check_output(['py', env_arg],
```

Comment:
It looks like we lost `--version` in the most recent commit here. When testing this, the script hangs indefinitely.

Reply:
Oups ! Indeed. I corrected the second step and broke the third... Corrected.